api/opds.py

```diff
@@ -176,9 +176,28 @@ def visible_delivery_mechanisms(self, licensepool):
             yield lpdm
 
     def annotate_work_entry(self, work, active_license_pool, edition, identifier, feed, entry, updated=None):
+        # If ElasticSearch included a more accurate last_update_time,
+        # use it instead of Work.last_update_time
+        updated = work.last_update_time
+        if isinstance(work, WorkSearchResult):
+            # Elasticsearch puts this field in a list, but we've set it up
+            # so there will be at most one value.
+            last_updates = getattr(work._hit, 'last_update', [])
+            if last_updates:
+                # last_update is seconds-since epoch; convert to UTC datetime.
+                updated = datetime.datetime.utcfromtimestamp(last_updates[0])
+
+                # There's a chance that work.last_updated has been
+                # modified but the change hasn't made it to the search
+                # engine yet. Even then, we stick with the search
+                # engine value, because a sorted list is more
+                # important to the import process than an up-to-date
+                # 'last update' value.
+
```

Comment:
This was previously in `CirculationManagerAnnotator`, which meant the functionality wasn't available for `CrawlableCollectionLane` -- last update time was being calculated and ignored.